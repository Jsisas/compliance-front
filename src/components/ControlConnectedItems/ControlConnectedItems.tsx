import * as React from 'react';
import { Col, Row, Typography } from 'antd';
import { Link } from 'react-router-dom';
import styles from './controlConnectedItems.module.scss';
import { Requirement } from '../../redux/Requirement/RequirementSlice';

const { Text } = Typography;

interface TaskConnectedItemsProps {
	requirements?: Requirement[];
}

export function ControlConnectedItems(props: TaskConnectedItemsProps): JSX.Element {
	return (
		<>
			<Col xs={24} className={styles.connectedItem}>
				<Row gutter={[16, 16]}>
					<Col xs={24}>
						<Text style={{ fontWeight: 600 }}>Connected items</Text>
					</Col>
				</Row>
				<Row gutter={[16, 0]}>
					<Col xs={24}>
						<Text type={'secondary'}>Requirements ({props.requirements?.length})</Text>
					</Col>
				</Row>
				<Row gutter={[16, 16]}>
					<Col xs={24} data-testid={'list'}>
						{(props.requirements || []).length > 0 ? (
							props.requirements?.map((requirement) => {
								return (
									<Link key={requirement.id} to={`/requirements/${requirement.id}`} style={{ display: 'block' }}>
										{requirement.title}
									</Link>
								);
							})
						) : (
							<span>No connected requirements</span>
						)}
					</Col>
				</Row>
			</Col>
		</>
	);
}
