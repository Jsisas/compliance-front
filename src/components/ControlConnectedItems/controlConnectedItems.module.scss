.connectedItem{
  background: #FFFFFF;
  border: 1px solid #DCDDDD;
  box-sizing: border-box;
  border-radius: 4px;

  a{
    padding-bottom: 8px;
  }
}
