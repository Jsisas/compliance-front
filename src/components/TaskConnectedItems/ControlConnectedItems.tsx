import * as React from 'react';
import {Col, Row, Typography} from "antd";
import {Link} from "react-router-dom";
import {Regulation} from "../../redux/Regulation/RegulationSlice";
import {Task} from "../../redux/Task/TaskSlice";
import {Requirement} from "../../redux/Requirement/RequirementSlice";
import {Control} from "../../redux/Control/ControlSlice";
import {lowerCameltoUpperCamel, shortenStringLength} from "../../util/StringUtil";
import styles from './contorlConnectedItems.module.scss';

const {Text} = Typography;

interface TaskConnectedItemsProps {
    task: Task;
}

export function TaskConnectedItems(props: TaskConnectedItemsProps) {

    return (
        <>
            <Col xs={24} className={styles.connectedItem}>
                <Row gutter={[16, 16]}>
                    <Col xs={24}>
                        <Text style={{fontWeight: 600}}>Connected items</Text>
                    </Col>
                </Row>
                <Row gutter={[16, 0]}>
                    <Col xs={24}>
                        <Text type={'secondary'}>Control</Text>
                    </Col>
                </Row>
                <Row gutter={[16, 16]}>
                    <Col xs={24}>
                        <Link key={props.task.control.id} to={`/controls/${props.task.control.id}`}>
                            {shortenStringLength(props.task.control.title, 50)}
                        </Link>
                    </Col>
                </Row>
            </Col>
        </>
    );
};