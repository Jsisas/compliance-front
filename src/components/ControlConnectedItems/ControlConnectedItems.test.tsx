import React from 'react';
import { ControlConnectedItems } from './ControlConnectedItems';
import { deepRender } from '../../util/test/TestUtil';
import { screen } from '@testing-library/react';
import { testRequirements } from '../../util/test/data/DataUtil';

describe('<ControlConnectedItems />', () => {
	it('it matches a snapshot', () => {
		const [history, element] = deepRender(<ControlConnectedItems requirements={testRequirements} />);
		expect(element).toMatchSnapshot();
	});

	it('it shows all the requirements', () => {
		deepRender(<ControlConnectedItems requirements={testRequirements} />);

		const requirementsCount = screen.getByText(`Requirements (${testRequirements.length.toString()})`);
		const list = screen.getByTestId('list');

		expect(requirementsCount).toBeInTheDocument();
		expect(list.children.length).toBe(testRequirements.length);
	});

	it('it shows fallback text', () => {
		deepRender(<ControlConnectedItems requirements={[]} />);

		const requirementsCount = screen.getByText(`Requirements (${0})`);
		const list = screen.getByTestId('list');
		const fallbackText = screen.getByText('No connected requirements');

		expect(requirementsCount).toBeInTheDocument();
		expect(fallbackText).toBeInTheDocument();
		expect(list.children.length).toBe(1);
	});
});
