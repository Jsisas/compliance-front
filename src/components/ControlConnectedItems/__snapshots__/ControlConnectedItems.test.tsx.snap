// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlConnectedItems /> it matches a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-col connectedItem ant-col-xs-24"
      >
        <div
          class="ant-row"
          style="margin: -8px -8px 8px -8px;"
        >
          <div
            class="ant-col ant-col-xs-24"
            style="padding: 8px 8px 8px 8px;"
          >
            <span
              class="ant-typography"
              style="font-weight: 600;"
            >
              Connected items
            </span>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -8px; margin-right: -8px;"
        >
          <div
            class="ant-col ant-col-xs-24"
            style="padding-left: 8px; padding-right: 8px;"
          >
            <span
              class="ant-typography ant-typography-secondary"
            >
              Requirements (
              4
              )
            </span>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin: -8px -8px 8px -8px;"
        >
          <div
            class="ant-col ant-col-xs-24"
            data-testid="list"
            style="padding: 8px 8px 8px 8px;"
          >
            <a
              href="/requirements/e6d9d11e-f662-4102-a48c-8eaa1e4dade3"
              style="display: block;"
            >
              Equipment maintenance
            </a>
            <a
              href="/requirements/005034f1-c4f6-4b2b-8825-175c1d5de2f9"
              style="display: block;"
            >
              Classification of information
            </a>
            <a
              href="/requirements/979a092e-2b5b-4fa9-858d-641104dc9d85"
              style="display: block;"
            >
              Policies for information security
            </a>
            <a
              href="/requirements/0f7d4d87-47fa-4d07-9dc4-f55ba22229d7"
              style="display: block;"
            >
              Review of Information Security Policy
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-col connectedItem ant-col-xs-24"
    >
      <div
        class="ant-row"
        style="margin: -8px -8px 8px -8px;"
      >
        <div
          class="ant-col ant-col-xs-24"
          style="padding: 8px 8px 8px 8px;"
        >
          <span
            class="ant-typography"
            style="font-weight: 600;"
          >
            Connected items
          </span>
        </div>
      </div>
      <div
        class="ant-row"
        style="margin-left: -8px; margin-right: -8px;"
      >
        <div
          class="ant-col ant-col-xs-24"
          style="padding-left: 8px; padding-right: 8px;"
        >
          <span
            class="ant-typography ant-typography-secondary"
          >
            Requirements (
            4
            )
          </span>
        </div>
      </div>
      <div
        class="ant-row"
        style="margin: -8px -8px 8px -8px;"
      >
        <div
          class="ant-col ant-col-xs-24"
          data-testid="list"
          style="padding: 8px 8px 8px 8px;"
        >
          <a
            href="/requirements/e6d9d11e-f662-4102-a48c-8eaa1e4dade3"
            style="display: block;"
          >
            Equipment maintenance
          </a>
          <a
            href="/requirements/005034f1-c4f6-4b2b-8825-175c1d5de2f9"
            style="display: block;"
          >
            Classification of information
          </a>
          <a
            href="/requirements/979a092e-2b5b-4fa9-858d-641104dc9d85"
            style="display: block;"
          >
            Policies for information security
          </a>
          <a
            href="/requirements/0f7d4d87-47fa-4d07-9dc4-f55ba22229d7"
            style="display: block;"
          >
            Review of Information Security Policy
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
